### PR TITLE
Update to python 3.8.2

### PR DESCRIPTION
**Pull request by misl6**
Changelog:

- updates `python3` recipe version to `3.8.2`.
- adds `hostopenssl` recipe
- builds the `hostpython3` recipe w/ `ssl` support (via `hostopenssl`)
- `python3 toolchain.py pip install packagename` now uses `pip3` provided by`hostpython3`
- updates `cookiecutter` in order to support `python3.8.2` (adds `LC_CTYPE=UTF-8` for locale encoding, adds recursive search for `python3` headers, fixes runtime `subprocess` imports) 
--- EDIT ---
- updates `kivy` to a post `2.0.0rc1`, see comment on file for checking out why.

**TODO**

- [x] Use hostpython `pip` for installing plain python packages (needs ssl)
- [x] Update cookiecutter template
- [x] Cleanup
- [x] Test release on testflight

### CODE REVIEW

File: recipes/hostpython3/__init__.py [hunk 1]
```diff
@@ -28,33 +28,11 @@ def prebuild_arch(self, arch):
         if self.has_marker("patched"):
             return
         self.copy_file("ModulesSetup", "Modules/Setup.local")
-        # self.apply_patch("ssize-t-max.patch")
-        # self.apply_patch("dynload.patch")
-        # self.apply_patch("static-_sqlite3.patch")
-        # shutil.copy("Modules/Setup.dist", "Modules/Setup")
-        # if "openssl.build_all" in self.ctx.state:
-        #     self.append_file("ModulesSetup.openssl", "Modules/Setup.local")
```
Comment (AndreMiras) [inline recipes/hostpython3/__init__.py:36]: I haven't played with kivy-ios much, but just curious why we have theses commented lines?
Wouldn't it be cleaner to just drop it?

File: recipes/hostpython3/__init__.py [hunk 2]
```diff
@@ -28,33 +28,11 @@ def prebuild_arch(self, arch):
...
         self.set_marker("patched")
 
     def postbuild_arch(self, arch):
         return
```
Comment (AndreMiras) [inline recipes/hostpython3/__init__.py:34]: Same here with the dead code and commented lines, how about dropping it if we don't use?

File: recipes/hostpython3/__init__.py (hunk 1 above)
Comment (misl6) [reply recipes/hostpython3/__init__.py:36]: I don't understand, I already removed these lines.
Probably was some test that maybe @tito left over here last time he did a fix

File: recipes/hostpython3/__init__.py (hunk 2 above)
Comment (misl6) [reply recipes/hostpython3/__init__.py:34]: Same as before

File: recipes/hostpython3/__init__.py (hunk 1 above)
Comment (AndreMiras) [reply recipes/hostpython3/__init__.py:36]: Yes I actually brain farted as I thought you added them :facepalm: 
I guess I hat too much of programming this weekend :smile: 
Thanks for the clean up :clap:

File: recipes/hostpython3/__init__.py (hunk 2 above)
Comment (AndreMiras) [reply recipes/hostpython3/__init__.py:34]: Yes never mind, I brain farted big time :smile:

**Pull request merged**